packages/calypso-e2e/src/lib/pages/editor-page.ts

```diff
@@ -368,7 +294,7 @@ export class EditorPage {
 		openInlineInserter: OpenInlineInserter
 	): Promise< ElementHandle > {
 		// First, launch the inline inserter in the way expected by the script.
-		await openInlineInserter( this.editorWindow );
+		await openInlineInserter( await this.editor.canvas() );
```

Comment:
**comment** - I see what you mean in your [response](https://github.com/Automattic/wp-calypso/pull/76085#issuecomment-1525938711) to my (admittedly) poorly-phrased comment:

> I might be missing something, but I don't think that a consumer forgetting to call this method would be a case because otherwise, they won't be able to interact with the editor. The parent has to be retrieved before interacting with the Editor/canvas.

Instead of placing the burden of retrieving the editor and canvas, you've built it into the individual methods in `EditorPage`. So as long as developers implementing new methods for EidtorPage calls the relevant methods to retrieve the editor/canvas, things will work just fine.

Reply:
Indeed! The only difference is that the editor/canvas locator must now be awaited, but it still returns a locator. The need to explicitly specify whether we want to work with the editor's parent (`await this.editor.parent()`) or the canvas (`await this.editor.canvas()`) remains unchanged, as we cannot reliably predict if the editor/canvas will be iframed or not.